lib/assessment.js

```diff
@@ -29,7 +46,11 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+var array = [];
+  for (var i = 0; i < length; i++) {
+ // again I am unsure here how to pass the defaultFunction index of the array
+  };
+  return array;
```

Comment:
You did most of the tricky stuff like in Q3, but sat out the part that makes this function most interesting. 1/2